pkg/etcd/etcd.go

```diff
@@ -480,3 +550,67 @@ func (e *ETCD) clientURLs(ctx context.Context, clientAccessInfo *clientaccess.In
 	}
 	return clientURLs, memberList, nil
 }
+
+func (e *ETCD) Snapshot(ctx context.Context) {
+	ticker := time.NewTicker(e.config.SnapshotInterval)
+	defer ticker.Stop()
+	for snapshotTime := range ticker.C {
+		logrus.Infof("Snapshot retention check")
+		snapshotDir, err := snapshotDir(e.config)
+		if err != nil {
+			logrus.Errorf("failed to get the snapshot dir: %v", err)
+			continue
+		}
+		if err := snapshotRetention(e.config.SnapshotRetention, snapshotDir); err != nil {
+			logrus.Errorf("failed to apply snapshot retention: %v", err)
+			continue
+		}
+		logrus.Infof("Taking etcd snapshot at %s", snapshotTime.String())
+		sManager := snapshot.NewV3(nil)
+		tlsConfig, err := toTLSConfig(e.runtime)
+		if err != nil {
+			logrus.Errorf("failed to get tls config for etcd: %v", err)
+			continue
+		}
+		etcdConfig := etcd.Config{
+			Endpoints: []string{"https://127.0.0.1:2379"},
+			TLS:       tlsConfig,
+			Context:   ctx,
+		}
+		snapshotPath := filepath.Join(snapshotDir, "etcd-snapshot"+strconv.Itoa(int(snapshotTime.Unix())))
+
+		if err := sManager.Save(ctx, etcdConfig, snapshotPath); err != nil {
+			logrus.Errorf("failed to save snapshot %s: %v", snapshotPath, err)
+			continue
+		}
+	}
+}
+
+func snapshotRetention(retention int, snapshotDir string) error {
+	snapshotFiles := []os.FileInfo{}
+	if err := filepath.Walk(snapshotDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if strings.HasPrefix(info.Name(), etcdSnapshotPrefix) {
+			snapshotFiles = append(snapshotFiles, info)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	if len(snapshotFiles) <= retention {
+		return nil
+	}
+	sort.Slice(snapshotFiles, func(i, j int) bool {
+		return snapshotFiles[i].ModTime().Before(snapshotFiles[j].ModTime())
```

Comment:
Should this be sorting based on creation time?

Reply:
I can see that fileinfo interface has only modTime() function, is there a way to get creation time for a file other than modtime?